src/convertToObject.js

```diff
@@ -5,8 +5,23 @@
  *
  * @return {object}
  */
+
 function convertToObject(sourceString) {
-  // write your code here
+  const parts = sourceString.split(';');
+  const result = {};
+
+  for (const part of parts) {
+    const piece = part.trim().split(':');
+
+    if (piece.length === 2) {
+      const key = piece[0].trim();
+      const value = piece[1].trim();
+
+      result[key] = value;
+    }
+  }
+
+  return result;
```

Comment:
The function `convertToObject` uses a `for...of` loop to iterate over the parts of the string, which is not recommended according to the checklist. Instead, consider using array iteration methods like `map`, `filter`, and `reduce` to transform the string into an object. This approach would be more declarative and in line with functional programming practices.